geode-core/src/main/java/org/apache/geode/cache/client/proxy/ProxySocketFactories.java

```diff
@@ -14,19 +14,29 @@
  */
 package org.apache.geode.cache.client.proxy;
 
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_ENABLED_COMPONENTS;
+
+import java.util.Properties;
+
 import org.apache.geode.cache.client.SocketFactory;
+import org.apache.geode.security.SecurableCommunicationChannels;
+
 
 public class ProxySocketFactories {
 
   /**
    * Create a {@link SocketFactory} that will connect a geode client through
    * the configured SNI proxy.
    *
-   *
    * @param hostname the hostname of the sni proxy
    * @param port the port of the sni proxy
    */
-  public static SocketFactory sni(String hostname, int port) {
+  public static SocketFactory sni(String hostname, int port, Properties geodeProperties) {
```

Comment:
I am really not keen on using `Properties` configure proxies. How does the user know which arbitrary `Properties` object to pass? A strongly typed value expresses the correct value without support documentation. Also are we saying that the user is responsible for passing the configuration rather than the system passing it on the `createSocket` method or some lifecycle method?

(Javadocs are missing)